dask/bytes/core.py

```diff
@@ -194,63 +117,165 @@ def read_bytes(urlpath, delimiter=None, not_zero=False, blocksize=2**27,
+                                                text, encoding)
     except KeyError:
-        raise NotImplementedError("Unknown protocol for reading %s (%s)" %
+        raise NotImplementedError("Unknown protocol %s (%s)" %
                                   (protocol, urlpath))
-
-    return read_bytes(storage_options.pop('path'), delimiter=delimiter,
-                      not_zero=not_zero, blocksize=blocksize, sample=sample,
-                      compression=compression, **storage_options)
+    return fs, myopen
 
 
-def open_files_by(open_files_backend, path, compression=None, **kwargs):
-    """ Given open files backend and path return dask.delayed file-like objects
-
-    NOTE: This is an internal helper function, please refer to
-    :func:`open_files` documentation for more details.
+def open_files(urlpath, compression=None, mode='rb', encoding='utf8',
+               errors=None, name_function=None, num=1, **kwargs):
+    """ Given path return dask.delayed file-like objects
 
     Parameters
     ----------
-    path: string
-        Filepath or globstring
+    urlpath: string
+        Absolute or relative filepath, URL (may include protocols like
+        ``s3://``), or globstring pointing to data.
     compression: string
         Compression to use.  See ``dask.bytes.compression.files`` for options.
+    mode: 'rb', 'wt', etc.
+    encoding: str
+        For text mode only
+    errors: None or str
+        Passed to TextIOWrapper in text mode
+    name_function: function or None
+        if opening a set of files for writing, those files do not yet exist,
+        so we need to generate their names by formatting the urlpath for
+        each sequence number
+    num: int [1]
+        if writing mode, number of files we expect to create (passed to
+        name+function)
     **kwargs: dict
         Extra options that make sense to a particular storage connection, e.g.
         host, port, username, password, etc.
 
+    Examples
+    --------
+    >>> files = open_files('2015-*-*.csv')  # doctest: +SKIP
+    >>> files = open_files('s3://bucket/2015-*-*.csv.gz', compression='gzip')  # doctest: +SKIP
+
     Returns
     -------
     List of ``dask.delayed`` objects that compute to file-like objects
     """
-    files = open_files_backend(path, **kwargs)
-
-    if compression:
-        decompress = merge(seekable_files, compress_files)[compression]
-        if PY2:
-            files = [delayed(SeekableFile)(file) for file in files]
-        files = [delayed(decompress)(file) for file in files]
-
-    return files
+    fs, paths, myopen = get_fs_paths_myopen(urlpath, compression, mode,
+                                            encoding=encoding, num=num,
+                                            name_function=name_function,
+                                            **kwargs)
+    if 'w' in mode:
+        return [delayed(myopen, pure=False)(path, mode) for path in
+                paths]
```

Comment:
FWIW I tend to do the line break on the for

```python
return [delayed(myopen, pure=False)(path, mode) 
        for path in paths]
```

It feels nicer to keep the clauses together.